src/Compilers/Core/Portable/Syntax/SyntaxNode.cs

```diff
@@ -421,7 +421,7 @@ public bool ContainsDiagnostics
         /// </summary>
         public bool Contains(SyntaxNode node)
         {
-            if (node == null || !node.FullSpan.IntersectsWith(node.FullSpan))
+            if (node == null || !this.FullSpan.IntersectsWith(node.FullSpan))
```

Comment:
I agree `Contains` implies parent/child, I'd missed that aspect. Given that and the potential performance difference @jcouv mentioned, it makes sense to leave the `while` loop as is.